nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -82,10 +84,16 @@
 
     public static final PropertyDescriptor LINE_SPLIT_COUNT = new PropertyDescriptor.Builder()
             .name("Line Split Count")
-            .description("The number of lines that will be added to each split file")
+            .description("The number of lines that will be added to each split file, excluding header lines")
             .required(true)
             .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
             .build();
+    public static final PropertyDescriptor FRAGMENT_MAX_SIZE = new PropertyDescriptor.Builder()
+            .name("Maximum Fragment Size")
+            .description("The maximum size of each split file, including header lines")
```

Comment:
Seems like there could be some interesting edge conditions.  For instance what if a given line is by itself larger than the max?  Or, does the max get exceeded by the size of at most one line then split or is the max exceeded then a pushback mechanism goes to the previous line and does the split and then continues?  I might be overthinking this but in any case would be good to let the user know how these edge cases are handled.